collector/internal/controller.go

```diff
@@ -42,6 +44,9 @@ func NewController(cfg *controller.Config,
 	}
 	cfg.Reporter = rep
 
+	// Provide internal metrics via the collectors telemetry.
+	otel.SetMeterProvider(rs.MeterProvider)
```

Comment:
I think we may assume that the rest of the collector did not set a global meter provider. Is there a reason for that? I'm wondering if we're going to be negatively affecting the rest of the collector (or invalidating assumptions made by other collector subsystems) by setting a global meter provider here.

Note that we don't have to set a global meter provider, we can propagate this specific meter provider to the point where it's instantiated by the metrics logic.